File: app/src/main/java/com/gasparaiciukas/owntrainer/utils/NutrientValueFormatter.java
package com.gasparaiciukas.owntrainer.utils;

import com.github.mikephil.charting.data.PieEntry;
import com.github.mikephil.charting.formatter.ValueFormatter;

import java.text.DecimalFormat;

public class NutrientValueFormatter extends ValueFormatter {
    @Override
    public String getPieLabel(float value, PieEntry pieEntry) {
        pieEntry.setLabel("");
        return super.getPieLabel(value, pieEntry);
    }

    @Override
    public String getFormattedValue(float value) {
        DecimalFormat df = new DecimalFormat("#");
        String floatToPercent = df.format(value);
        if (value == 0) return "";
        else return floatToPercent + "%";
    }
}

File: app/src/main/java/com/gasparaiciukas/owntrainer/utils/NutrientValueFormatter.kt
package com.gasparaiciukas.owntrainer.utils

import com.github.mikephil.charting.data.PieEntry
import com.github.mikephil.charting.formatter.ValueFormatter
import java.text.DecimalFormat

class NutrientValueFormatter : ValueFormatter() {
    override fun getPieLabel(value: Float, pieEntry: PieEntry): String {
        pieEntry.label = "" // remove labels
        return super.getPieLabel(value, pieEntry)
    }

    override fun getFormattedValue(value: Float): String {
        val df = DecimalFormat("#")
        val floatToPercent = df.format(value.toDouble())
        return if (value == 0f) "" else "$floatToPercent%"
    }
}